core/templates/domain/exploration/read-only-exploration-backend-api.service.spec.ts

```diff
@@ -95,139 +72,150 @@ describe('Read only exploration backend API service', function() {
       version: 1,
       state_classifier_mapping: {}
     };
-  }));
+  });
 
-  afterEach(function() {
-    $httpBackend.verifyNoOutstandingExpectation();
-    $httpBackend.verifyNoOutstandingRequest();
+  afterEach(() => {
+    httpTestingController.verify();
   });
 
   it('should successfully fetch an existing exploration from the backend',
-    function() {
-      var successHandler = jasmine.createSpy('success');
-      var failHandler = jasmine.createSpy('fail');
+    fakeAsync(() => {
+      let successHandler = jasmine.createSpy('success');
+      let failHandler = jasmine.createSpy('fail');
 
-      $httpBackend.expect('GET', '/explorehandler/init/0').respond(
-        sampleDataResults);
-      ReadOnlyExplorationBackendApiService.fetchExploration(
+      readOnlyExplorationBackendApiService.fetchExploration(
         '0', null).then(successHandler, failHandler);
-      $httpBackend.flush();
+      var req = httpTestingController.expectOne(
+        '/explorehandler/init/0');
+      expect(req.request.method).toEqual('GET');
+      req.flush(sampleDataResults);
+
+      flushMicrotasks();
 
-      expect(successHandler).toHaveBeenCalledWith(sampleDataResults);
+      expect(successHandler).toHaveBeenCalled();
       expect(failHandler).not.toHaveBeenCalled();
-    }
+    })
   );
 
   it('should load a cached exploration after fetching it from the backend',
-    function() {
-      var successHandler = jasmine.createSpy('success');
-      var failHandler = jasmine.createSpy('fail');
+    fakeAsync(() => {
+      let successHandler = jasmine.createSpy('success');
+      let failHandler = jasmine.createSpy('fail');
 
       // Loading a exploration the first time should fetch it from the backend.
-      $httpBackend.expect('GET', '/explorehandler/init/0').respond(
-        sampleDataResults);
-      ReadOnlyExplorationBackendApiService.loadExploration(
-        '0', null).then(successHandler, failHandler);
-      $httpBackend.flush();
+      readOnlyExplorationBackendApiService.loadLatestExploration(
+        '0').then(successHandler, failHandler);
+      var req = httpTestingController.expectOne(
+        '/explorehandler/init/0');
+      expect(req.request.method).toEqual('GET');
+      req.flush(sampleDataResults);
 
-      expect(successHandler).toHaveBeenCalledWith(sampleDataResults);
+      flushMicrotasks();
+
+      expect(successHandler).toHaveBeenCalled();
       expect(failHandler).not.toHaveBeenCalled();
 
       // Loading a exploration the second time should not fetch it.
-      ReadOnlyExplorationBackendApiService.loadExploration(
-        '0', null).then(successHandler, failHandler);
+      readOnlyExplorationBackendApiService.loadLatestExploration(
+        '0').then(successHandler, failHandler);
 
-      expect(successHandler).toHaveBeenCalledWith(sampleDataResults);
+      expect(successHandler).toHaveBeenCalled();
```

Comment:
Can you make it `toHaveBeenCalledWith` like before?

Reply:
Done